kernel/src/actions/deletion_vector.rs

```diff
@@ -223,15 +232,37 @@ pub(crate) fn treemap_to_bools(treemap: RoaringTreemap) -> Vec<bool> {
     }
 }
 
+#[derive(Debug, Clone)]
+#[repr(C)]
+pub struct DeletionVector {
```

Comment:
I think we should try and keep all `#[repr(C)]` stuff in the `ffi` crate.

Since this is really just a wrapper around the treemap could we:
1. Get rid of it here, and just directly use the TreeMap when needed
2. Make a wrapper struct like this in the `ffi` crate

Perhaps we'll want to expand this struct at some point, so I could see arguing to keep it here as well, but I think for now we could remove it and add it back if we find a use case.

Reply:
I removed it